src/modules/advancedOrders/hooks/useGetQuote.ts

```diff
@@ -0,0 +1,66 @@
+import { useLayoutEffect } from 'react'
+import { useSetAtom } from 'jotai'
+import { onlyResolvesLast } from 'legacy/utils/async'
+
+import { getQuote } from 'api/gnosisProtocol'
+import { OrderQuoteResponse } from '@cowprotocol/cow-sdk'
+import GpQuoteError from 'api/gnosisProtocol/errors/QuoteError'
+import { useQuoteParams } from './useQuoteParams'
+import { advancedOrdersQuoteAtom } from '../state/advancedOrdersQuoteAtom'
+import { useAdvancedOrdersFullState } from './useAdvancedOrdersFullState'
+import { useUpdateCurrencyAmount } from './useUpdateCurrencyAmount'
+import { Field } from 'legacy/state/swap/actions'
+
+// Every 10s
+const PRICE_UPDATE_INTERVAL = 10_000
+
+// Solves the problem of multiple requests
+const getQuoteOnlyResolveLast = onlyResolvesLast<OrderQuoteResponse>(getQuote)
+
+// TODO: this also should be unified for each trade widget
+export function useGetQuote() {
```

Comment:
This hook + updater should be refactored to fit the architecture:
https://github.com/cowprotocol/cowswap/blob/e9b841f090ad75d4540d2402b64a544e309ee338/src/cow-react/modules/trade/docs/abstraction.drawio.svg